modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLSSOForAdminLoginTestCase.java

```diff
@@ -0,0 +1,258 @@
+import org.wso2.identity.integration.test.util.Utils;
+import org.wso2.identity.integration.test.utils.CommonConstants;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.xpath.XPathExpressionException;
+
+/**
+ * This class contains test cases for SAML SSO for Admin Console.
+ */
+public class SAMLSSOForAdminLoginTestCase extends AbstractSAMLSSOTestCase {
+
+    private static final Log log = LogFactory.getLog(SAMLSSOForAdminLoginTestCase.class);
+    private ServerConfigurationManager serverConfigurationManager;
+    private final SAMLConfig samlConfig = new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, null, null,
+            null, null);
+
+    // Constants.
+    private static final String SAML_ACS_URL = "https://localhost:9853/acs";
+    private static final String SAML_SSO_URL = "https://localhost:9853/samlsso";
+    private static final String MANAGEMENT_CONSOLE_LOGIN_URL = "https://localhost:9853/carbon/admin/login.jsp";
+    private static final String MANAGEMENT_CONSOLE_LOGOUT_URL = "https://localhost:9853/carbon/admin/logout_action.jsp";
+    private static final String MANAGEMENT_CONSOLE_HOME_URL = "https://localhost:9853/carbon/admin/index.jsp";
+    private static final String AUTHENTICATION_PORTAL_LOGIN_URL =
+            "https://localhost:9853/authenticationendpoint/login.do";
+    private static final String USERNAME = "admin";
+    private static final String PASSWORD = "admin";
+
+    // File paths.
+    private final String carbonHome = Utils.getResidentCarbonHome();
+    private final String serverConfigFilePath = buildPath(getISResourceLocation(), "saml",
+            "saml-sso-for-admin-console.toml");
+    private final String SAMLSSOSpXmlPath = buildPath(FrameworkPathUtil.getSystemResourceLocation(), "artifacts",
+            "IS", "saml", "filebasedspidpconfigs", "self.xml");
+    private final String identityConfigSAMLSSOSpXmlPath = buildPath(carbonHome, "repository", "conf",
+            "identity", "service-providers", "self.xml");
+    private final String identityConfigPath = buildPath(carbonHome, "repository", "conf", "identity",
+            "service-providers");
+    private final String ssoIdPConfigXmlPath = buildPath(carbonHome, "repository", "conf", "identity",
+            "sso-idp-config.xml");
+    private final String ssoIdPConfigXmlToCopyPath = buildPath(FrameworkPathUtil.getSystemResourceLocation(),
+            "artifacts", "IS", "saml", "filebasedspidpconfigs", "management-console-sso-idp-config.xml");
+    private final String ssoIdPConfigXmlOriginalConfigPath = buildPath(FrameworkPathUtil.getSystemResourceLocation(),
+            "artifacts", "IS", "saml", "filebasedspidpconfigs", "original-sso-idp-config.xml");
+
+    public SAMLSSOForAdminLoginTestCase() {
+
+        if (log.isDebugEnabled()) {
+            log.info(String.format("SAML SSO for Admin Console Test initialized for %s", SAML_SSO_URL));
```

Comment:
We can replace String.format with a concatenation for better performance.

Reply:
Since this is a test case, we have to consider the readability of the code more than the performance. So I believe current approach needs not to be changed.